streaming-types/src/time_conversions.rs

```diff
@@ -1,24 +1,53 @@
 use crate::frame_metadata_v1_generated::GpsTime;
+use anyhow::{anyhow, Error, Result};
 use chrono::{DateTime, Datelike, NaiveDate, Timelike, Utc};
 
-impl From<GpsTime> for DateTime<Utc> {
-    fn from(t: GpsTime) -> Self {
+impl TryFrom<GpsTime> for DateTime<Utc> {
+    fn try_from(t: GpsTime) -> Result<Self> {
+        if t.nanosecond() > 999 {
+            return Err(anyhow!("Timestamp Error ns = {0} > 999", t.nanosecond()));
+        }
+        if t.microsecond() > 999 {
+            return Err(anyhow!("Timestamp Error: us = {0}", t.microsecond()));
+        }
+        if t.millisecond() > 999 {
+            return Err(anyhow!("Timestamp Error: ms = {0}", t.millisecond()));
+        }
         let nanosecond = (t.millisecond() as u32 * 1_000_000)
             + (t.microsecond() as u32 * 1_000)
             + (t.nanosecond() as u32);
 
-        NaiveDate::from_yo_opt(2000 + (t.year() as i32), t.day().into())
-            .expect("Date should be valid")
-            .and_hms_nano_opt(
-                t.hour().into(),
-                t.minute().into(),
-                t.second().into(),
-                nanosecond,
-            )
-            .expect("Time should be valid")
-            .and_local_timezone(Utc)
-            .unwrap()
+        let dt = match NaiveDate::from_yo_opt(2000 + (t.year() as i32), t.day().into()) {
```

Comment:
`Option::ok_or` does this is less steps: https://doc.rust-lang.org/std/option/enum.Option.html#method.ok_or